### DISCUSSION TITLE
Mesh Generation with Complex Fault Structures

### DISCUSSION THREAD

**Discussion by FURUNZ** [SeisSol/SeisSol#1222] [General]
I am currently working on simulating the dynamic rupture of a 3D fault, and I have encountered some difficulties during the modeling process. I have referred to the 3D fault modeling method in TPV29, where I first use GMSH to generate a 2D mesh, and then use a .f90 executable to modify the fault elevation. Another method I am referencing is based on a published paper using SeisSol, where the fault geometry is generated from the fault trace file (fault_trace.pl).
However, I am facing challenges in implementing the fault simulation process. Could you provide a more detailed reference process? Specifically, I have studied the examples in SeisSol and the training material, but most of the cases involving complex fault modeling provide the mesh directly, without detailing the mesh generation process. I would like to know if it is possible to achieve this using SimModeler and GOCAD as alternatives, similar to the method used in the example where GMSH was applied.
In summary, I have thoroughly studied the cases in SeisSol and examples in the training materials. For complex fault modeling, their meshes are mostly provided directly, without a detailed process on how to generate the mesh. I was wondering if you could offer any more detailed references or processes for mesh generation.
Thank you very much!

**Comment by FURUNZ** [comment]
SimModeler, Gacad is not available directly from the official website,When I refer to the URL in mannul
https://github.com/SeisSol/Meshing/tree/master/SimModelerDownloadingBuilding

**Reply by Thomas-Ulrich** [reply]
simmodeler is free to use for academic use (appart from a yearly report). But you need to ask them directly for a license.
An example of workflow is described here:
https://seissol.readthedocs.io/en/latest/simmodelerCAD-workflow.html
